tests/middleware.test.js

```diff
@@ -284,9 +284,9 @@ describe('validate body middleware', () => {
 
       await new Promise((resolve, reject) => {
         socket.on('connect', () => reject('Should not be able to connect'));
-        socket.on('error', error => {
-          if (error === 'Authentication error') resolve();
-          else reject(new Error('Wrong error: ' + error));
+        socket.on('connect_error', error => {
+          if (error instanceof Error) resolve();
```

Comment:
You should probably still test the error message too. In theory, an error with a different message could be passed in.

Reply:
Aha so `error.message === 'Authentication error' && error instanceof Error`?

Reply:
No, the other order. If this were TypeScript, we'd want `error` to be typed as `unknown`. If `error instanceof Error` returns `true`, then the type is narrowed to `Error`, and it's only at that point that we know that it *has* a `.message` property for us to check.

Reply:
No, the other order. If this were TypeScript, we'd want `error` to be typed as `unknown`. If `error instanceof Error` returns `true`, then the type is narrowed to `Error`, and it's only at that point that we know that it *has* a `.message` property for us to check.